portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java

```diff
@@ -364,19 +364,22 @@ public DLFileVersion cancelCheckOut(long userId, long fileEntryId)
 	}
 
 	@Override
-	public void checkFileEntries(long checkInterval) throws PortalException {
+	public void checkFileEntries(long companyId, long checkInterval)
+		throws PortalException {
+
 		Date date = new Date();
 
-		if (_previousCheckDate == null) {
-			_previousCheckDate = new Date(
-				date.getTime() - (checkInterval * Time.MINUTE));
+		if (_companyPreviousCheckDate.get(companyId) == null) {
```

Comment:
To avoid race conditions (there should be none, but ...) can you use `putIfAbsent` here?

Reply:
done, thanks!!!